src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -254,22 +254,33 @@ protected Task Stop()
                     [String]$CommandName
                 )
                 try {
-                    $null = Microsoft.PowerShell.Core\Get-Command $CommandName -ErrorAction Stop
+                    $command = Microsoft.PowerShell.Core\Get-Command $CommandName -ErrorAction Stop
                 } catch [System.Management.Automation.CommandNotFoundException] {
                     $PSCmdlet.ThrowTerminatingError($PSItem)
                 }
                 try {
-                    $null = Microsoft.PowerShell.Core\Get-Help $CommandName -Online
-                } catch [System.Management.Automation.PSInvalidOperationException] {
-                    Microsoft.PowerShell.Core\Get-Help $CommandName -Full
-                }";
+                    $helpUri = [Microsoft.PowerShell.Commands.GetHelpCodeMethods]::GetHelpUri($command)
+                    # HEAD means we don't need the content itself back :)
+                    $status = (Invoke-WebRequest -Method Head -Uri $helpUri -ErrorAction Stop).StatusCode
+                    if ($status -lt 400)
+                    {
+                        $null = Microsoft.PowerShell.Core\Get-Help $CommandName -Online
+                        return
+                    }
+                } catch {
+                    # Ignore - we want to drop out to Get-Help -Full
+                }
+                return Microsoft.PowerShell.Core\Get-Help $CommandName -Full
+                ";
 
             if (string.IsNullOrEmpty(helpParams)) { helpParams = "Get-Help"; }
 
             PSCommand checkHelpPSCommand = new PSCommand()
                 .AddScript(CheckHelpScript, useLocalScope: true)
                 .AddArgument(helpParams);
 
+            // TODO: Rather than print the help in the console, we should send the string back
```

Comment:
This actually fits nicely into the changes that https://github.com/powershell/vscode-powershell/issues/884 morphed into.